core/src/main/java/com/sequenceiq/cloudbreak/service/altus/AltusMachineUserService.java

```diff
@@ -176,14 +178,19 @@ public DataBusCredential getOrCreateDataBusCredentialIfNeeded(Long stackId) thro
     }
 
     public CdpAccessKeyType getCdpAccessKeyType(StackDto stackDto) {
-        if (!entitlementService.isECDSABasedAccessKeyEnabled(stackDto.getAccountId())) {
-            return CdpAccessKeyType.ED25519;
-        }
-        Optional<Image> image = componentConfigProviderService.findImage(stackDto.getId());
-        if (image.isEmpty()) {
-            return CdpAccessKeyType.ED25519;
+        if (AwsConstants.AwsVariant.AWS_NATIVE_GOV_VARIANT.variant().value().equals(stackDto.getPlatformVariant())) {
+            Optional<Image> image = componentConfigProviderService.findImage(stackDto.getId());
+            if (image.isEmpty()) {
+                throw new CloudbreakRuntimeException("We can't figure out package versions for image because the image is empty.");
+            } else {
+                if (telemetryFeatureService.isECDSAAccessKeyTypeSupported(image.get().getPackageVersions())) {
+                    return CdpAccessKeyType.ECDSA;
+                } else {
+                    throw new CloudbreakRuntimeException("The image contains packages which can't support ECDSA key, but ECDSA is mandatory on Gov environment");
```

Comment:
It should not happen in any case (as GOV images should support it) and I think it would be too much additional work to move this to the stack creation phase. I think it's not worth moving this there.